services/r2ctl/config/r2ctl.go

```diff
@@ -52,31 +54,41 @@ type Configuration struct {
 	Metrics instrument.MetricsConfiguration `yaml:"metrics"`
 
 	// Store configuration.
-	Store R2StoreConfiguration `yaml:"store"`
+	Store r2StoreConfiguration `yaml:"store"`
 
 	// Simple Auth Config.
 	Auth *auth.SimpleAuthConfig `yaml:"auth"`
-
-	// Clock configuration used by the api.
-	Clock clock.Configuration `yaml:"clock"`
 }
 
-// R2StoreConfiguration has all the fields necessary for an R2Store
-type R2StoreConfiguration struct {
+// r2StoreConfiguration has all the fields necessary for an R2 store.
+type r2StoreConfiguration struct {
```

Comment:
so if anyone wants to run a different main, they have to completely redefine all of these configs?

Reply:
The idea is that the users wouldn't need to run their own main file if the config here is flexible enough, and I'd rather keep it that way.